client/landing/stepper/README.md

```diff
@@ -1,106 +1,402 @@
+  - [Reusability](#reusability)
+  - [Renaming steps](#renaming-steps)
+- [State management](#state-management)
+  - [Typed state](#typed-state)
+  - [Miscellaneous fields](#miscellaneous-fields)
+- [Creating a site](#creating-a-site)
+- [The API](#the-api)
+- [Useful utilities and hooks](#useful-utilities-and-hooks)
+- [Troubleshooting](#troubleshooting)
+- [Help and feedback](#help-and-feedback)
 
-### Example flow
+## How does it work
+
+A Stepper flow **is not a series of steps**; it's a graph of steps with a non-deterministic order. The first step of the flow is deterministic, but the following steps are decided on the fly by the flow depending on state (user input, user status, site status, etc...).
+
+An ideal Stepper flow is a [finite-state machine](https://en.wikipedia.org/wiki/Finite-state_machine): it has a pre-defined set of possible states, and it should only be in one of those states at any given moment.
+
+### The most important principle: Steps shouldn't make decisions or communicate
+
+![StepperFlow](https://github.com/user-attachments/assets/c8d5b07a-4172-4ba1-92d0-7821f0b6b9f0)
+
+Stepper steps should be a lot like native form inputs. They may receive some properties, they may have internal state (think checkbox), and they must submit some information. That's it.
+
+1. They should not make narrative decisions (go back, go next, exit flow, etc..)
+2. They should not communicate with other steps (e.g. via query params).
+3. **They should not have side effects (e.g. persisting stuff in local storage).**
+
+They should **only submit** to the flow, and the flow should do all the thinking, persisting, and the navigation.
+
+**Common question**: What if a step wants to cancel/skip/drop out? Then it should submit something like `{ action: skip }`.
+
+<details><summary><strong>Expand if you're curious why.</strong></summary>
+<p>
+	<ol>
+		<li>When steps make navigational decisions, the aforementioned finite-state machine is thrown out of the window. The dependency graph explodes and things get out of control quickly. Especially if you consider that steps are reusable across flows.</li>
+		<li>Same thing happens when steps communicate: the flow is blind to some decisions, and the steps become less re-usable.</li>
+		<li>When a step has side effects in Flow A, they might affect the step's behaviour in Flow B. Only the flow should persist things.</li>
+	</ol>
+</p>
+</details>
+
+In general, the smarter the step, the more problematic and tailored it is. Please treat your steps as buttons or inputs.
+
+### Making a flow
+
+A flow is a collection of steps. Each of these steps submit some information to the flow, which means the state of the flow is largely the sum of these submitted data. For that reason, the `FlowV2` interface requires the steps collection to be defined before the flow itself. This way, the flow can shape its state around the submissions and properties of these steps.
```

Comment:
We could also mention:
- making sure that flow IDs are unique (and step IDs too)
- not to create flows by extending (object.assign) other flows, as that makes it really difficult to determine the flow behaviour (some aspects are overridden, others are preserved from the original flow)